packages/flutter/lib/src/material/text_field.dart

```diff
@@ -569,6 +569,11 @@ class _TextFieldState extends State<TextField> with AutomaticKeepAliveClientMixi
     assert(debugCheckHasDirectionality(context));
     final ThemeData themeData = Theme.of(context);
     final TextStyle style = widget.style ?? themeData.textTheme.subhead;
+
+    if (style.inherit == false && (style.fontSize == null || style.textBaseline == null)) {
```

Comment:
This should be an assert and it would be OK to assume that textTheme.subhead is fully specified. That way, if the assert fails, it's clear that it's about the style value provided to the TextField